Sources/Headers.swift

```diff
@@ -49,9 +43,9 @@ extension Headers {
         return headers.isEmpty
     }
 
-    public subscript(name: HeaderName) -> HeaderValues {
+    public subscript(name: CaseInsensitiveString) -> Header {
```

Comment:
Can we allow them to set a `HeaderConvertible` object.

I want to allow this type of syntax:

```swift
let key: String = "key"
let value: String = "value"
request.headers[key] = value
request.headers.append(value)
```